src/Fantomas.Core/CodePrinter.fs

```diff
@@ -300,14 +300,27 @@ let genInheritConstructor (ic: InheritConstructor) =
         genType node.Type
         +> sepSpaceOrIndentAndNlnIfExpressionExceedsPageWidth (genExpr node.Expr)
 
-let isSynExprLambdaOrIfThenElse (e: Expr) =
+let genParenExpr openingParen e closingParen r =
```

Comment:
`gen` is a misleading name here.
You are constructing a new ExprParenNode here, not generating new events inside the `Context`.
Maybe go with some like `mkExprParenNode`. Just avoid `gen` here.